src/documentation-helpers.ts

```diff
@@ -190,13 +187,19 @@ export const depictReadonly: Depicter<z.ZodReadonly<z.ZodTypeAny>> = ({
   next,
 }) => next({ schema: schema._def.innerType });
 
+/** @since OAS 3.1 nullable replaced with type arrays */
 export const depictNullable: Depicter<z.ZodNullable<z.ZodTypeAny>> = ({
   schema,
   next,
-}) => ({
-  nullable: true,
-  ...next({ schema: schema.unwrap() }),
-});
+}) => {
+  const nested = next({ schema: schema.unwrap() });
+  return {
+    ...nested,
+    type: ["null" as Extract<SchemaObjectType, string>].concat(
+      "$ref" in nested ? [] : nested.type || [],
+    ),
```

Comment:
this should be extracted and reused along with `onEach` handling of actually nullable stuff